plugin/cache/cache.go

```diff
@@ -109,8 +110,9 @@ func (w *ResponseWriter) WriteMsg(res *dns.Msg) error {
 	if key != -1 && duration > 0 {
 		w.set(res, key, mt, duration)
 
-		cacheSize.WithLabelValues(Success).Set(float64(w.pcache.Len()))
-		cacheSize.WithLabelValues(Denial).Set(float64(w.ncache.Len()))
+		l := strings.Join(w.Zones, ",")
```

Comment:
This might be pretty expensive and create zone gc, is there a way around it?

Reply:
Why are we including this in the metrics?

Reply:
See the commit message and pr description.

Reply:
This can be written GC free by currying metrics. I'll look into it.